File: diquis-local-frontend/src/types/account.ts
export interface Account {
    id: string;
    user_id: string;
    name: string;
    account_type: 'checking' | 'saving' | 'cash' | 'digital';
    currency: string;
    current_balance: number;
    allow_negative_balance: boolean;
    theme_color: string;
    is_system: boolean;
    is_active: boolean;
    created_at: string;
}

export interface AccountCreatePayload {
    name: string;
    account_type: 'checking' | 'saving' | 'cash' | 'digital';
    currency: string;
    initial_balance?: number;
    allow_negative_balance: boolean;
    theme_color: string;
}

export interface AccountUpdatePayload {
    name?: string;
    is_active?: boolean;
    theme_color?: string;
    allow_negative_balance?: boolean;
}

File: diquis-local-backend/app/schemas/account.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import datetime

class AccountBase(BaseModel):
    name: str
    account_type: str = Field(..., pattern="^(checking|saving|cash|digital)$")
    currency: Optional[str] = Field(None, min_length=3, max_length=3, description="Código ISO de 3 letras (ej. USD, CRC)")
    allow_negative_balance: Optional[bool] = False
    theme_color: Optional[str] = 'silver'
    is_system: Optional[bool] = False

class AccountCreate(AccountBase):
    initial_balance: Optional[float] = 0.0

class AccountResponse(AccountBase):
    id: UUID
    user_id: UUID
    current_balance: float
    is_active: bool
    created_at: datetime

    class Config:
        from_attributes = True

class AccountUpdate(BaseModel):
    name: Optional[str] = None
    is_active: Optional[bool] = None
    theme_color: Optional[str] = None
    allow_negative_balance: Optional[bool] = None
    currency: Optional[str] = Field(None, min_length=3, max_length=3)